Yoast/wordpress-seo
# Rename alert to notification

## Context


The term `Alert` was used in the code to describe `Notifications`. To avoid confusion, especially with the introduction of Alert boxes (#14585), all "alerts" referring to notifications were renamed. 

## Summary


This PR can be summarized in the following changelog entry:

* N/A

## Relevant technical choices:

* Renames all "alert" (classes, variables, etc.) to "notification",
* Renames the following files:
  * `class-yoast-alerts.php`
  * `partial-alerts-template.php`
  * `partial-alerts-warnings.php`
  * `partial-alerts-errors.php`
  * alerts.scss 
* Renames `.yoast-alert`, `.yoast-alerts`, `.yoast-alert-holder`, `.yoast-alert-header`, `.yoast-alert-active`, `.yoast-alert-dismissed`

**Note**: `.yoast-alert-box` in `css/src/admin-global.scss` is __not__ renamed, since this is used for an actual alert box (see image below). I will look into this in issue #14585. 
<img width="665" alt="yoast-alert-box" src="https://user-images.githubusercontent.com/43582255/77751064-4b70b580-7025-11ea-9e20-93b7b0114c26.png">

* Also removes the unused method `show_overview_page()` (with permission from Omar).

## Test instructions

This PR can be tested by following these steps:

* Checkout this branch and build,
* In the backend, go to `Settings` > `Reading, check the box for `Search Engine Visibility` and save changes,
* See a popup appear in the admin bar "There is a new notification" and an extra count (see image below),
  <img width="288" alt="Screenshot 2020-03-27 at 12 28 34" src="https://user-images.githubusercontent.com/43582255/77751708-80313c80-7026-11ea-826c-579dfef140c7.png">
* Go to `SEO` > `General` and see the following problem,
  <img width="541" alt="problems-notifications" src="https://user-images.githubusercontent.com/43582255/77752177-5b899480-7027-11ea-817f-1d226dd75739.png">
* Check whether you can hide the notification by clicking on the eye, 
  <img width="524" alt="hidden-blog-public-notice" src="https://user-images.githubusercontent.com/43582255/77752016-0a79a080-7027-11ea-8e10-11bb90f50b3d.png">
* And check whether you can make it un-hidden by clicking on the eye again.

All should work and look as it does in trunk. 
Any additional ideas for testing this PR are appreciated :).


## UI changes

## Documentation

## Quality assurance


Fixes #14619
